rfc/0006-threat-model.md

```diff
@@ -0,0 +1,262 @@
+|              |                                  |
+| :----------- | :------------------------------- |
+| Feature Name | Kubernetes admission control threat model                           |
+| Start Date   | 04/18/2022                          |
+| Category     | Security                       |
+| RFC PR       | https://github.com/kubewarden/rfc/pull/2  |
+
+# Summary
+[summary]: #summary
+
+Describe the threat model for the admissions control, how it can effect Kubewarden
+and the possible mitigations.
+
+# Motivation
+[motivation]: #motivation
+
+The goal of this is turn Kubewarden mode secure to operate.
+
+## Examples / User Stories
+[examples]: #examples
+
+As Kubewarden developer, I want to know which are the weak point of the Kubernetes admission control feature, so I can
+mitigate this problem.
+
+As Kubewarden user, I want to know which are the weak points of the Kubernetes admission control, so I can apply the mitigation
+in my cluster.
+
+# Detailed design
+[design]: #detailed-design
+
+The threat model of the Kubernetes admission control list 18 possible threats.
+All of them will be discussed sharing a brief description of it, possible mitigation
+and what we, as Kubewarden team, should do about it (if necessary). As well as
+help our users to be aware of the risks.
+
+If more details about each threat is needed, go to the original document here:
+https://github.com/kubernetes/sig-security/tree/main/sig-security-docs/papers/admission-control
+
+
+## Threat 1 - Attacker floods webhook with traffic preventing its
+
+An attacker who has access to the Webhook endpoint, at the network level, could
+send large  quantities of traffic, causing an effective denial of service to the
+admission controller.
+
+**Mitigation**
+
+Webhook fails closed. In other words, if the webhook does not respond in time,
+for any reasion, API server should reject the request. We are safe on this.
+Kubewarden default behavior  already does that.
+
+## Threat 2 - Attacker passes workloads which require complex  processing causing timeouts
+
+An attacker, who can access the admission controller at a network level, passes
+requests to  the admission controller which require complex processing, causing
+timeouts as the  admission controller uses compute power to process the workloads
+
+**Mitigation**
+Webhook fail closed and authenticate callers. We are safe, Kubewarden already does that.
```

Comment:
just to mention that someone could change `failurePolicy` to `ignore`. But by default is `fail` so I agree that we are safe here.

Reply:
Same comment as in Threat 1.

Reply:
Also in this case, I think the only reasonable way to mitigate this issue is to introduce allow each policy to run for at most X seconds. Once the timeout is reached the evaluation response will be a rejection accompanied by a certain error code and message.